docs/03-deploy.Rmd

```diff
@@ -178,3 +178,74 @@ deploy:
 This means the website will be pushed to the `master` branch of the repository `username/username.github.io` (remember to replace `username` with your actual username).
 
 You can also deploy your website to Amazon S3\index{Amazon S3}, and the setup on the R side is very similar to what we have introduced for GitHub Pages. The only difference is in the last step, where you change the target from GitHub Pages to Amazon S3. For more information, please see the documentation on Travis: https://docs.travis-ci.com/user/deployment/s3/.
+
+## GitLab Pages
+
+[GitLab](http://gitlab.com/) is a very popular way to host the source code of your project.
+GitLab has a [built in Continuous Integration & Deployment service](https://about.gitlab.com/features/gitlab-ci-cd/)
+that can be used to host static websites, named [GitLab Pages](https://about.gitlab.com/features/pages/).
+The major advantage of using GitLab Pages is that
+GitLab Continuous Integration & Deployment service will compile all your Rmd posts
+instead of your local computer
+**and any generated content, such as HTML files, will be stored in the Git repository**.
+
+Before I show you how, I’d like to mention two issues that you should be aware of:
+
+- It is somewhat unpredictable the output
+  because it is fully automatic
+  and you do not have a chance to see the new content or results to be published.
+  There are many factors that could affect building the site:
+  the R version,
+  availability of certain R packages,
+  system dependencies, and network connection, etc.
+- The time required to compile all Rmd files may be very long
+  and cause timeouts on GitLab,
+  depending on how time-consuming your R code is.
+
+The first problem can be solved by specifying the version of the package you are using
+on the installation instructions.
+The second problem can be solved by using caching mechanism.
```

Comment:
Please remove lines 192-207, and simply mention something like `this approach has similar issues as we mentioned in Section \@ref(travis-github)`, instead of repeating them in this section.